qfieldsync/core/cloud_api.py

```diff
@@ -733,6 +943,29 @@ def _clear_cloud_cookies(self, url: QUrl) -> None:
         for cookie in self._nam.cookieJar().cookiesForUrl(url):
             self._nam.cookieJar().deleteCookie(cookie)
 
+    def _add_csrf_cookies_to_request(self, url: QUrl, request: QNetworkRequest) -> None:
+        """
+        Adds CSRF cookie to a network request.
+        Required for some QFieldCloud requests, like POST/PUT/PATCH/DELETE.
+        """
+        cookie_jar = self._nam.cookieJar()
+        cookies = cookie_jar.cookiesForUrl(url)
+
+        csrftoken_cookie = next(
+            (cookie for cookie in cookies if cookie.name() == b"csrftoken"), None
+        )
+
+        if csrftoken_cookie:
+            request.setRawHeader(b"X-CSRFToken", csrftoken_cookie.value())
+            request.setRawHeader(b"Referer", url.toEncoded())
+
+    def _add_provider_id_header_to_request(self, request: QNetworkRequest) -> None:
+        """Adds the current provider id to a request, if any. Sometimes required by QFC middleware."""
+        provider_id = self.auth_provider_id()
+
+        if provider_id and len(provider_id) > 0:
+            request.setRawHeader(b"X-QFC-IDP-ID", QByteArray(provider_id.encode()))
```

Comment:
Do not hardcode the header. Best case should come from QFC, worst case should be a const.